Code/IO/src/sitkImageViewer.cxx

```diff
@@ -152,107 +169,127 @@ void ImageViewer::initializeDefaults()
+      m_DefaultSearchPath.push_back ( ProgramFiles + "\\" );
       }
     }
 
   if ( itksys::SystemTools::GetEnv ( "USERPROFILE", ProgramFiles ) )
     {
-    m_GlobalDefaultSearchPath.push_back ( ProgramFiles + "\\" );
-    m_GlobalDefaultSearchPath.push_back ( ProgramFiles + "\\Desktop\\" );
+    m_DefaultSearchPath.push_back ( ProgramFiles + "\\" );
+    m_DefaultSearchPath.push_back ( ProgramFiles + "\\Desktop\\" );
     }
 
 #elif defined(__APPLE__)
 
   // Common places on the Mac to look
-  m_GlobalDefaultSearchPath = { "/Applications/", "/Developer/", "/opt/", "/usr/local/" };
+  m_DefaultSearchPath = { "/Applications/", "/Developer/", "/opt/", "/usr/local/" };
   std::string homedir;
   if ( itksys::SystemTools::GetEnv ( "HOME", homedir ) )
     {
-    m_GlobalDefaultSearchPath.push_back( homedir + "/Applications/" );
+    m_DefaultSearchPath.push_back( homedir + "/Applications/" );
     }
 
 
 #else
 
   // linux and other systems
-  m_GlobalDefaultSearchPath = { "./" };
+  m_DefaultSearchPath = { "./" };
   std::string homedir;
   if ( itksys::SystemTools::GetEnv ( "HOME", homedir ) )
     {
-    m_GlobalDefaultSearchPath.push_back( homedir + "/bin/" );
+    m_DefaultSearchPath.push_back( homedir + "/bin/" );
     }
 
-  m_GlobalDefaultSearchPath.emplace_back("/opt/" );
-  m_GlobalDefaultSearchPath.emplace_back("/usr/local/" );
+  m_DefaultSearchPath.emplace_back("/opt/" );
+  m_DefaultSearchPath.emplace_back("/usr/local/" );
 
 #endif
 
-  localDebugMacro( << "Default search path: " << m_GlobalDefaultSearchPath << std::endl );
-
   //
   //  Set the ExecutableNames
   //
 #if defined(_WIN32)
-  m_GlobalDefaultExecutableNames = { "Fiji.app/ImageJ-win64.exe", "Fiji.app/ImageJ-win32.exe" };
+  m_DefaultExecutableNames = { "Fiji.app/ImageJ-win64.exe", "Fiji.app/ImageJ-win32.exe" };
 #elif defined(__APPLE__)
-  m_GlobalDefaultExecutableNames = { "Fiji.app" };
+  m_DefaultExecutableNames = { "Fiji.app" };
 #else
   // Linux
-  m_GlobalDefaultExecutableNames = { "Fiji.app/ImageJ-linux64", "Fiji.app/ImageJ-linux32" };
+  m_DefaultExecutableNames = { "Fiji.app/ImageJ-linux64", "Fiji.app/ImageJ-linux32" };
 #endif
 
 #ifdef _WIN32
-  m_GlobalProcessDelay = 1;
+  m_ProcessDelay = 1;
 #else
-  m_GlobalProcessDelay = 500;
+  m_ProcessDelay = 500;
 #endif
 
-  ImageViewer::m_GlobalDefaultApplication = ImageViewer::FindViewingApplication();
 
-  m_GlobalViewerImageCount = 0;
-  m_GlobalAreDefaultsInitialized = true;
+  m_DefaultApplication = FindViewingApplication();
   }
 
-
-//
-// Constructor
-//
-ImageViewer::ImageViewer()
+GlobalConfig& GlobalConfig::getInstance()
   {
+  static GlobalConfig instance;
+  static bool first_time = true;
 
-  if (!m_GlobalAreDefaultsInitialized)
+  if (first_time)
     {
-    initializeDefaults();
+    if (instance.m_DefaultDebug)
+      {
+      std::ostringstream msg;
+      msg << instance;
+      ::itk::OutputWindowDisplayDebugText( msg.str().c_str() );
+      }
+    first_time = false;
```

Comment:
Please use std::call_once with a lambda to avoid a race condition with first_time.